codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
@nmittler - I think breaking the `goingAway()` event up into sent/received makes sense and avoids some confusion that recent PRs have helped address.

> To change goingAway to be onGoAwaySent and make the inbound GOAWAY exclusive to Http2FrameListener.onGoAwayRead

I don't think it is ideal but given the current structure of our interfaces I think this may make sense.  I don't think a `Http2Connection.Listener.goAwayRecieved(..)` provides any additional information, and it seems like it is called in exactly the same situations as `Http2FrameListener.onGoAwayRead`. We should also be consistent with our use of the `on` prefix from the `Http2Connection.Listener`  (i.e. either use it everywhere or drop it).